lib/puppet/type/openldap_database.rb

```diff
@@ -56,6 +56,9 @@
       when 'RedHat'
         if Facter.value(:operatingsystemmajrelease).to_i <= 6
           'bdb'
+        # # TODO: Remove legacy fact when drop Puppet 6 support
+        elsif Facter.value('os.release.major').to_i >= 9 || Facter.value(:operatingsystemmajrelease).to_i >= 9
```

Comment:
Why add the legacy facy here at all? os.release.major is available in Puppet 6 as well.

Reply:
I think you can do `Facter.value('os')['release']['major']` which should work on Puppet 6 and 7.